test/unit/db/repos/dapps.js

```diff
@@ -0,0 +1,357 @@
+					link: dapp.asset.dapp.link,
+					transactionId: dapp.id,
+				});
+
+				return expect(result).to.be.empty;
+			});
+		});
+
+		describe('list()', () => {
+			it('should use pass correct params to inline sql', function*() {
+				sinonSandbox.spy(db, 'query');
+				const params = {
+					limit: 10,
+					offset: 20,
+				};
+				yield db.dapps.list(params);
+
+				expect(db.query.firstCall.args[0]).to.be.a('string');
+				return expect(db.query.firstCall.args[1]).to.be.eql(params);
+			});
+
+			it('should rejected with error if required param "limit" is not passed', () => {
+				return expect(
+					db.dapps.list({
+						// limit: 10,
+						offset: 0,
+					})
+				).to.be.rejectedWith("Property 'limit' doesn't exist.");
+			});
+
+			it('should rejected with error if required param "offset" is not passed', () => {
+				return expect(
+					db.dapps.list({
+						limit: 10,
+						// offset: 0
+					})
+				).to.be.rejectedWith("Property 'offset' doesn't exist.");
+			});
+
+			it('should resolve if required params "limit" and "offset" is provided', () => {
+				return expect(
+					db.dapps.list({
+						limit: 10,
+						offset: 0,
+					})
+				).to.be.fulfilled;
+			});
+
+			it('should resolve with proper response', function*() {
+				const dapp = (yield seeder.seedDapps(db, 1))[0];
+				const result = yield db.dapps.list({ limit: 10, offset: 0 });
+
+				expect(result).to.be.not.empty;
+				expect(result).to.be.lengthOf(1);
+				expect(result[0]).to.have.all.keys(
+					'name',
+					'description',
+					'tags',
+					'link',
+					'type',
+					'category',
+					'icon',
+					'transactionId'
+				);
+				expect(result[0].transactionId).to.eql(dapp.id);
+				expect(result[0].name).to.eql(dapp.asset.dapp.name);
+				return expect(result[0].link).to.eql(dapp.asset.dapp.link);
+			});
+
+			it('should resolved with filtered results if param "where" is provided', function*() {
+				yield seeder.seedDapps(db, 1);
+				const result = yield db.dapps.list({
+					limit: 10,
+					offset: 0,
+					where: ["name = 'Unknown'"],
+				});
+
+				return expect(result).to.be.empty;
+			});
+
+			it('should resolved with ascending results if "sortField" and "sortMethod=ASC" is provided', function*() {
+				yield seeder.seedDapps(db, 5);
+				const result = yield db.dapps.list({
+					limit: 10,
+					offset: 0,
+					sortField: 'name',
+					sortMethod: 'ASC',
+				});
+
+				expect(result).to.be.lengthOf(5);
+				return expect(result).to.be.eql(_.orderBy(result, 'name'));
+			});
+
+			it('should resolved with descending results if "sortField" and "sortMethod=DESC" is provided', function*() {
+				yield seeder.seedDapps(db, 5);
+				const result = yield db.dapps.list({
+					limit: 10,
+					offset: 0,
+					sortField: 'name',
+					sortMethod: 'DESC',
+				});
+
+				expect(result).to.be.lengthOf(5);
+				return expect(result).to.be.eql(_.orderBy(result, 'name', 'desc'));
+			});
+
+			it('should paginate the result properly', function*() {
```

Comment:
can be fragile, as per other review

Reply:
Yes it may be fragile but with small set of data of 5 records to be tested, it is not behaving wrong.